alerter.c

```diff
@@ -1,29 +1,57 @@
 #include <stdio.h>
 #include <assert.h>
 
+#define PRODUCTION_CODE_ENABLE  STD_OFF 
+#define NETWORK_ALERT_THRESHOLDLEVEL 200
+
 int alertFailureCount = 0;
 
-int networkAlertStub(float celcius) {
+#if (PRODUCTION_CODE_ENABLE == STD_OFF)
+
+int networkAlertStub(float celcius )
+{   
     printf("ALERT: Temperature is %.1f celcius.\n", celcius);
     // Return 200 for ok
     // Return 500 for not-ok
-    // stub always succeeds and returns 200
+    // stub always succeeds and returns 200    
     return 200;
 }
 
-void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
+void networkAlert_test (float celcius)
+{
+    if (celcius < NETWORK_ALERT_THRESHOLDLEVEL)
+         {
+             assert(networkAlertStub(celcius)==200);
+         }
+    else

```

Comment:
Break the test into two. Keep tests simple and avoid if, for, etc in them. Complex tests themselves tend to have bugs!